api/handlers.go

```diff
@@ -40,17 +54,28 @@ func (api *API) userHandlerPOST(w http.ResponseWriter, req *http.Request, ps htt
 		WriteError(w, errors.New("Failed to parse multipart parameters."), http.StatusBadRequest)
 		return
 	}
-	email := (user.Email)(req.PostFormValue("email"))
-	if !email.Validate() {
-		WriteError(w, user.ErrInvalidEmail, http.StatusBadRequest)
+	email, err := database.NewEmail(req.PostFormValue("email"))
+	if err != nil {
+		WriteError(w, database.ErrInvalidEmail, http.StatusBadRequest)
```

Comment:
I think

```suggestion
		WriteError(w, err, http.StatusBadRequest)
```

is better, in case we ever add contexts in `NewEmail`.

Reply:
Sure, makes sense. It will also make it more like the other endpoints.

I'm a little divided on the idea of context here because we're sending this data to the user which might potentially be an attacker, so it might make sense to not disclose which execution path they are hitting. But this might come later, no need to focus on defense prematurely.